packages/firestore/test/integration/api/query.test.ts

```diff
@@ -2203,6 +2203,153 @@ apiDescribe('Queries', persistence => {
     }
   ).timeout('90s');
 
+  // TODO(b/291365820): Stop skipping this test when running against the
+  // Firestore emulator once the emulator is improved to include a bloom filter
+  // in the existence filter messages that it sends.
+  // eslint-disable-next-line no-restricted-properties
+  (USE_EMULATOR ? it.skip : it)(
+    'bloom filter can correctly handle updated and newly added documents',
+    async () => {
+      // Prepare the names and contents of the 100 documents to create.
+      const testDocs: { [key: string]: object } = {};
+      for (let i = 0; i < 100; i++) {
+        testDocs['doc' + (1000 + i)] = { key: 42, updated: false };
+      }
+
+      // Ensure that the local cache is configured to use LRU garbage
+      // collection (rather than eager garbage collection) so that the resume
+      // token and document data does not get prematurely evicted.
+      const lruPersistence = persistence.toLruGc();
+
+      return withRetry(async attemptNumber => {
+        return withTestCollection(lruPersistence, testDocs, async coll => {
+          // Run a query to populate the local cache with the 100 documents
+          // and a resume token.
+          const snapshot1 = await getDocs(
+            query(coll, where('updated', '==', false))
+          );
+          expect(snapshot1.size, 'snapshot1.size').to.equal(100);
+          const createdDocuments = snapshot1.docs.map(snapshot => snapshot.ref);
+
+          // Delete 50 docs,update 25 docs of the 100 documents, and add 100 new docs. Use a
+          // WriteBatch, to avoid affecting the local cache.
+          const deletedDocumentIds = new Set<string>();
+          const updatedDocumentIds = new Set<string>();
```

Comment:
Now that I think of it, this test should also update some documents in such a way that they still match the query.